src/main/java/hudson/remoting/Launcher.java

```diff
@@ -390,6 +396,29 @@ public void run() throws CmdLineException, IOException, InterruptedException {
             return;
         }
 
+        if (connectionTarget != null) {
+            initialize();
+            runAsTcpClient();
+        } else if (agentJnlpURL != null || !urls.isEmpty() || directConnection != null) {
```

Comment:
Fortunately this now has test coverage! (Though https://github.com/jenkinsci/bom/pull/2417 is of no help since the relevant tests require special setup to run.)